05/app.js

```diff
@@ -0,0 +1,61 @@
+class Student {
+    constructor(firstName, lastName) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.grades = {};
+    }
+
+    // Static method to normalize subject names
+    static normalizeSubject(subject) {
+        return subject.trim().toLowerCase() //this could be adjusted further if we decide to handle spaces in subject names differently, perhaps by adding: .replace(/\s+/g, '_')
+    }
+
+    // Static method to compute the average of an array of numbers.
+    static computeAverage(arr) {
+        if (!Array.isArray(arr) || arr.length === 0) return null;
+
+        const sum = arr.reduce((acc, num) => acc + num, 0);
+
+        return sum / arr.length;
+    }
+
+    addGrade(subject, grade) {
+        if (typeof subject !== 'string' || subject.trim() === '') {
+            throw new Error('Subject must be a non-empty string');
+        }
+        if (typeof grade !== 'number' || isNaN(grade)) {
+            throw new Error('Grade must be a valid number');
+        }
+
+        const subjectKey = Student.normalizeSubject(subject);
```

Comment:
Może zamiast pamiętać za każdym razem, aby "normalizować" subject, to napisać metodę `getGradesBySubject()` i tam wykonywać odpowiedniego sprawdzenia + normalizacji itp. - łatwiej będzie utrzymać porządek.